src/utils.rs

```diff
@@ -102,14 +102,9 @@ where
 }
 
 /// Creates a secure RNG from a seed.
-pub fn rng(seed: &[u8; RNG_SEED]) -> ChaCha12Rng {
-    let mut hash = Sha256::new();
-
-    hash.update(seed);
-    hash.update(b"RNG");
-
-    let hash = hash.finalize().into();
-    ChaCha12Rng::from_seed(hash)
+/// We should keep this above 64 to be secure
+pub fn rng(seed: &[u8; 32]) -> ChaCha12Rng {
+    ChaCha12Rng::from_seed(*seed)
```

Comment:
Not sure I understand this comment, in conjunction with the change to a 256 bit seed. Also, if you're copying `*seed`, there's no reason to take it as reference.